lib/ansible/modules/cloud/vmware/vmware_guest_find.py

```diff
@@ -90,17 +90,27 @@ def __init__(self, module):
 
     def getvm_folder_paths(self):
         results = []
+        vms = []
+
+        if self.uuid:
+            vm_obj = find_vm_by_id(self.content, vm_id=self.uuid, vm_id_type="uuid")
+            if vm_obj is None:
+                self.module.fail_json(msg="Failed to find the virtual machine with UUID : %s" % self.uuid)
+            vms = [vm_obj]
+
+        elif self.name:
+            objects = self.get_managed_objects_properties(vim_type=vim.VirtualMachine, properties=['name'])
+            for temp_vm_object in objects:
+                if len(temp_vm_object.propSet) != 1:
+                    continue
+                for temp_vm_object_property in temp_vm_object.propSet:
```

Comment:
Why are you looping to all properties? If the need is to support MoRef, then why not add a new field instead of reusing something that is clearly meant for the name (hence `vm.config.name` checked)

MoRef can not be changed or managed, it is an internal ID unique within vCenter, and to use it as reference is fine, but i'd never put that value in a `name` field..

Reply:
Sorry, but I am not following you here. Could you please elaborate more on this ?

Reply:
Why are we looping through the propSet and not using `vm.config.name` to match on the name property (as it was before)?

Reply:
Done.